src/ImageSharp/MetaData/Profiles/Exif/ExifWriter.cs

```diff
@@ -86,23 +84,51 @@ public byte[] GetData()
                 return null;
             }
 
-            length += 10 + 4 + 2;
+            if (includeExifIdCode)
+            {
+                // Exif Code (6 bytes) + byte order marker (4 bytes)
+                length += 10;
+            }
+            else
+            {
+                // special case for PNG eXIf Chunk:
+                // two bytes for the byte Order marker 'II', followed by the number 42 (0x2A) and a 0, making 4 bytes total
+                length += 4;
+
+                // if the Exif Code ("Exif00") is not included, the start index is 0 instead of 6
+                startIndex = 0;
+            }
+
+            length += 4 + 2;
 
             byte[] result = new byte[length];
 
-            result[0] = (byte)'E';
-            result[1] = (byte)'x';
-            result[2] = (byte)'i';
-            result[3] = (byte)'f';
-            result[4] = 0x00;
-            result[5] = 0x00;
-            result[6] = (byte)'I';
-            result[7] = (byte)'I';
-            result[8] = 0x2A;
-            result[9] = 0x00;
-
-            int i = 10;
-            uint ifdOffset = ((uint)i - StartIndex) + 4;
+            int i = 0;
+            if (includeExifIdCode)
+            {
+                result[0] = (byte)'E';
```

Comment:
You could also increment `i` when you set the values so you don't need the duplicate code block in the else part.

Reply:
i have changed that